### ISSUE TITLE
Is it possible to add the feature to switch the dark mode automatically?

### ISSUE THREAD

**Issue by RaymondKJ** [ThomasFKJorna/zotero-night#13]
Hey, 

This plugin is very useful! Thanks for your great job!

My problem is that when I open Zotero, the default display is dark mode, and even if I manually switch to light mode, the next time I restart Zotero, it changes to dark mode again. Is it possible to keep the settings after a restart, i.e., keep the last mode I choose?

Moreover, could you please consider adding a feature to switch the night mode automatically? I use [AutoDarkMode](https://github.com/AutoDarkMode/Windows-Auto-Night-Mode) to manage my system display mode. I think it would be more beneficial for Zotero Night to add a feature to follow the system settings or switch automatically at regular intervals.

Best wishes!

**Comment by ThomasFKJorna** [comment]
Thanks for the kind words!

Good point about the PDF not switching back, I though it did do that but maybe some minor update from Zotero broke that, I see now that it also does not work on my end.

Keeping settings after restart is indeed intended, just implemented poorly haha, so I will fix that!

Re auto switching: this will be hard, as I don't think there are any hooks I can use to detect system dark-mode from within Zotero without manually implementing these for every platform, which I don't have the time to do unfortunately.
Browsers can detect system dark mode via the css @prefers-dark-theme media query, but the version of Firefox zotero uses is _just_ too old to support that (Zotero uses 60, and firefox supports that media query since 64 😢)

What could be possible is maybe setting some timer or possibly determining your sunset/sunrise times (definitely some package which does this easily) and set dark mode based on that! 


If anyone has any ideas for other ways to detect system dark mode please let me know!

**Comment by RaymondKJ** [comment]
@ThomasFKJorna Thanks much for your consideration!

> What could be possible is maybe setting some timer or possibly determining your sunset/sunrise times (definitely some package which does this easily) and set dark mode based on that!

I think a timer setting is enough; It would be a pleasant surprise if switching by sunset/sunrise times could be achieved!

Unfortunately, I have no clue about this.  Looking forward to expert ideas.

**Comment by Ahrdie** [comment]
@ThomasFKJorna Thanks a lot for explaining the conditions and creating this useful plugin!

If the firefox version is to be updated in a foreseeable future, I would favor waiting and using the media query instead of zotero specific parameters.
Otherwise the points, at which the switch from light to dark mode happens could possibly be off and not synchronous.

**Comment by ThomasFKJorna** [comment]
@Ahrdie yeah it's probably pretty difficult to make that seamless. I have no idea if there are any plans to update the firefox version, I've heard that the Zotero devs have been considering moving to non-mozilla architechture (maybe electron) at some point, but since everything moves relatively slowly i personally wouldnt bet on that happening within a year, so a "shitty" version might be better than none. It will be trivial to implement once these media queries are available anyway, so we wouldn't lose anything

**Comment by tefkah** [comment]
This will be possible in the next version of Zotero, so I'll add it then!

https://www.zotero.org/support/dev/zotero_7_for_developers
